src/cts/src/HTreeBuilder.cpp

```diff
@@ -363,103 +346,463 @@ void adjustToplevelTopology(Point<double>& a,
+  if (adjustAlongBlockage(targetDist,
+                          currLoc,
+                          parentPoint,
+                          sinks,
+                          x1,
+                          y1,
+                          x2,
+                          y2,
+                          scalingFactor,
+                          bestLoc)) {
+    return bestLoc;
+  }
+
+  // try moving beyond blockage boundary
+  return adjustBeyondBlockage(
+      currLoc, parentPoint, targetDist, sinks, scalingFactor);
+}
+
+bool HTreeBuilder::adjustAlongBlockage(double targetDist,
+                                       Point<double> currLoc,
+                                       Point<double> parentPoint,
+                                       const std::vector<Point<double>>& sinks,
+                                       double x1,
+                                       double y1,
+                                       double x2,
+                                       double y2,
+                                       int scalingFactor,
+                                       Point<double>& bestLoc)
+{
+  Point<double> newLoc = currLoc;
+  // clang-format off
+  debugPrint(logger_, CTS, "legalizer", 3, "{} currDist={:0.3f} != "
+	     "targetDist={:0.3f}, adjustAlongBlockage...", currLoc,
+	     currLoc.computeDist(parentPoint), targetDist);
+  // clang-format on
+  double x = currLoc.getX();
+  double y = currLoc.getY();
+  double px = parentPoint.getX();
+  double py = parentPoint.getY();
+
+  Point<double> noBest(-1e6, -1e6);
+  bestLoc = noBest;
+  double bestSinkDist = 1e12;
+  double sinkDist = 0.0;
+  std::vector<Point<double>> candidates;
+
+  // move along y axis within blockage
+  double newX = x;
+  double newY = py - targetDist + abs(px - newX);
+  newLoc.setX(newX);
+  newLoc.setY(newY);
+  candidates.emplace_back(newLoc);  // trial y#1
+
+  newY = py + targetDist - abs(px - newX);
+  newLoc.setY(newY);
+  candidates.emplace_back(newLoc);  // trial y#2
+
+  // move along x axis within blockage
+  newY = y;
+  newX = px - targetDist + abs(py - newY);
+  newLoc.setX(newX);
+  newLoc.setY(newY);
+  candidates.emplace_back(newLoc);  // trial x#1
+
+  newX = px + targetDist - abs(py - newY);
+  newLoc.setX(newX);
+  candidates.emplace_back(newLoc);  // trial x#2
+
+  for (const Point<double>& candidate : candidates) {
+    checkLegalityAndCostSpecial(currLoc,
+                                candidate,
+                                parentPoint,
+                                targetDist,
+                                sinks,
+                                scalingFactor,
+                                x1,
+                                y1,
+                                x2,
+                                y2,
+                                bestLoc,
+                                sinkDist,
+                                bestSinkDist);
+  }
+
+  if (bestLoc != noBest) {
+    return true;
+  }
   return false;
```

Comment:
You could just return the comparison:
`return bestLoc != noBest`

Reply:
Done